test/query-builder.js

```diff
@@ -0,0 +1,284 @@
+ /**
+ * Copyright (c) 2017-present, Jembi Health Systems NPC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict'
+
+const tap = require('tap')
+
+require('./test-env/init')()
+const config = require('../lib/config')
+const constants = require('../lib/constants')
+const queryBuilder = require('../lib/fhir/query-builder')()
+
+const FHIR_VERSION = config.getConf('server:fhirVersion')
+const searchParamsMap = require(`../lib/fhir/definitions/${FHIR_VERSION}/search-parameters-map.json`)
+
+const executeQueryBuilderTest = (t, functionNameToExecute, resourceType, queryParam, expectedResponse, value, desc) => {
+  t.test(desc, (t) => {
+    const searchParamsResource = searchParamsMap[resourceType]
+
+    const paramPath = searchParamsResource[queryParam].path
+    const propertyDefObj = queryBuilder.private.walkPropertyDef(paramPath, resourceType)
+    const modifier = constants.NO_MODIFER
+    const dateClause = queryBuilder.private[functionNameToExecute](paramPath, propertyDefObj.types[0], propertyDefObj, modifier, value)
+
+    t.ok(dateClause)
+    t.deepEqual(dateClause, expectedResponse, 'Should have the correct mongo clause')
+    t.end()
+  })
+}
+
+tap.test('.buildQueryForNumber should return a error (Not yet supported)', (t) => {
```

Comment:
Your test naming seems a bit inconsistent. Perhaps group everything under the name of the function they test `tap.test('.buildQueryForNumber')` and then for each test case add a `t.test('...case')`, I'm not sure the `t.test` and `desc` in the `executeQueryBuilderTest` function is necessary. Seems repetitive of the test name.